Cms/ProfileBioEdit.ascx.cs

```diff
@@ -0,0 +1,2394 @@
+        {
+            get { return Session["FamilyMembers"] as List<PersonFamilyMember>; }
+            set { Session["FamilyMembers"] = value; }
+        }
+
+        #region Base Control Methods
+
+        /// <summary>
+        /// Raises the <see cref="E:System.Web.UI.Control.Init" /> event.
+        /// </summary>
+        /// <param name="e">An <see cref="T:System.EventArgs" /> object that contains the event data.</param>
+        protected override void OnInit( EventArgs e )
+        {
+            base.OnInit( e );
+
+            this.BlockUpdated += Block_BlockUpdated;
+            this.AddConfigurationUpdateTrigger( upnlCareEntry );
+
+            BuildForm();
+        }
+
+        /// <summary>
+        /// Raises the <see cref="E:System.Web.UI.Control.Load" /> event.
+        /// </summary>
+        /// <param name="e">The <see cref="T:System.EventArgs" /> object that contains the event data.</param>
+        protected override void OnLoad( EventArgs e )
+        {
+            base.OnLoad( e );
+
+            lbSave.ValidationGroup = BlockValidationGroup;
+
+            lbCancel.ValidationGroup = BlockValidationGroup;
+            lbCancel.CausesValidation = false;
+
+            var displayButtonsInPanels = GetAttributeValue( AttributeKey.DisplayButtonsInAllPanels ).AsBoolean();
+
+            lbSave.Visible = !displayButtonsInPanels;
+            lbCancel.Visible = !displayButtonsInPanels;
+
+            pnlProfilePanels.DefaultButton = lbSave.ID;
+        }
+
+        #endregion Base Control Methods
+
+        #region Events
+
+        /// <summary>
+        /// Handles the BlockUpdated event of the control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void Block_BlockUpdated( object sender, EventArgs e )
+        {
+            BuildForm();
+        }
+
+        /// <summary>
+        /// Handles the Click event of the lbSave control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
+        protected void lbSave_Click( object sender, EventArgs e )
+        {
+            if ( Page.IsValid )
+            {
+                var rockContext = new RockContext();
+                var personGuid = hfPrimaryPersonGuid.Value.AsGuid();
+                var groupId = hfGroupId.Value.AsIntegerOrNull();
+
+                if ( !groupId.HasValue )
+                {
+                    // GroupId wasn't specified due to invalid situation
+                    // Return and report nothing.
+                    return;
+                }
+
+                var group = new GroupService( rockContext ).Get( groupId.Value );
+                if ( group == null )
+                {
+                    return;
+                }
```

Comment:
This code appears in multiple places within methods. Would it be worth doing this once onint or onpage load instead?

Reply:
Changed to global method instead, everywhere else but the Save method.